examples/java/unit-test/rest/TwilioRestTest.java

```diff
@@ -628,36 +710,121 @@ public void testShouldQueryParamInRequestwithIncorrectResponseCode() {
-        when(twilioRestClient.request(mockRequestPage0)).thenReturn(new Response(responseContent, 200));
+        Account.Status status = Account.Status.IN_PROGRESS;
+        Account account = new AccountUpdater(ACCOUNT_SID,status).update(twilioRestClient);
+    }
 
-        Request mockRequestPage1 = new Request(
-                HttpMethod.GET,
-                "api",
-                "/2010-04-01/Accounts.jsonFrom=9999999999&PageNumber=&To=4444444444&PageSize=2&Page=1&PageToken=PASMc49f620580b24424bcfa885b1f741130?PageSize=2"
+    @Test(expected = ApiException.class)
+    public void testShouldMakeInValidAPICallReturnsWrongStatusForAccountUpdater() {
+        Request mockRequest = new Request(
+                HttpMethod.POST,
+                com.twilio.rest.Domains.API.toString(),
+                "/2010-04-01/Accounts/AC222222222222222222222222222222.json"
         );
-        String responseContent2 = "{\"accounts\":[{\"call_sid\":\"PNXXXXY\", \"sid\":123, \"test_string\":\"Matey\"}], \"meta\": {\"url\":\"" + url + "\", \"next_page_url\":\"" + ""  + "\", \"previous_page_url\":\"" + url + "?PageSize=2" + "\", \"first_page_url\":\"" + url + "?PageSize=2" + "\", \"page_size\":2}}";
-        when(twilioRestClient.request(mockRequestPage1)).thenReturn(new Response(responseContent2, 200));
-        AccountReader accountReader = new AccountReader();
-        accountReader.setPageSize(2);
-        accountReader.previousPage(null, twilioRestClient);
+        mockRequest.addPostParam("PauseBehavior", "test");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        when(twilioRestClient.request(Mockito.any())).thenReturn(new Response("{\"account_sid\":\"AC222222222222222222222222222222\", \"call_sid\":\"PNXXXXY\"}", 404));
+        when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+        Account.Status status = Account.Status.IN_PROGRESS;
+        AccountUpdater accountUpdater = new AccountUpdater(ACCOUNT_SID,status);
+        accountUpdater.setPauseBehavior("test");
+        Account account = accountUpdater.update(twilioRestClient);
+    }
+
+    @Test
+    public void testShouldMakeValidAPICallAWSFetcher() {
+        String url = "https://api.twilio.com/v1/Credentials/AWS";
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        String testResponse =  "{\"credentials\":[{\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Ahoy\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}, {\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Hello\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}], \"meta\": {\"url\":\"" + url + "\", \"next_page_url\":\"" + url + "?PageSize=5" + "\", \"previous_page_url\":\"" + url + "?PageSize=3" + "\", \"first_page_url\":\"" + url + "?PageSize=1" + "\", \"page_size\":4}}";
```

Comment:
testResponse is not an Aws json string object, we need to change this.

Reply:
Done